lib/booker.rb

```diff
@@ -0,0 +1,95 @@
+require "time"
+require "date"
+
+require_relative "room"
+require_relative "reserve"
+require_relative "date_range"
+require_relative "blocks"
+
+class RoomBooker < Date
+  attr_reader :reservations, :rooms, :get_available_rooms
+
+  def initialize(rooms:)
+    @rooms = rooms
+    @reservations = []
+    @blocked_reservations = []
+  end
+
+  def book_reservation(check_in:, check_out:)
+    date_request = DateRange.new(check_in: check_in, check_out: check_out)
+    res_dates = date_request.dates_booked
+    find_room = find_available_room(check_in: check_in, check_out: check_out)
+
+    if find_room == nil
+      raise ArgumentError, "No rooms are available for these dates"
+    end
+
+    reservation_request = make_reservation(id: (@reservations.length + 1), dates_booked: res_dates, room_booked: find_room)
+
+    find_room.booked_on(check_in: check_in, check_out: check_out)
+    @reservations.push(reservation_request)
+    return reservation_request
+  end
+
+  def make_reservation(id:, dates_booked:, room_booked:)
+    reservation = Reservation.new(id: id, dates_booked: dates_booked, room_booked: room_booked)
+    return reservation
+  end
+
+  def find_room_id(id:)
+    return rooms.find { |room| room.id == id }
+  end
+
+  def find_available_room(check_in:, check_out:)
+    found_room = rooms.select { |room| room.room_available?(check_in: check_in, check_out: check_out) }
+    return found_room[0]
+  end
+
+  def date_query(date:)
+    find_dates = Date.parse(date)
+    find_by_dates = reservations.select { |reserved| reserved.dates_booked.include?(find_dates) == true }
```

Comment:
I don't like how these are named. `find_dates` implies more than one, and the query accepts one date.

`find_by_dates` sounds like a method to me, it implies a verb is happening. Better alternatives: `found_reservations` or `booking_list`